src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -6314,24 +6320,28 @@ MethodTableBuilder::PlaceMethodImpls()
             }
             else
             {
-                // Do not use pDecl->IsInterface here as that asks the method table and the MT may not yet be set up.
-                if (pCurDeclMethod->GetOwningType()->IsInterface())
-                {
-                    // Throws
-                    PlaceInterfaceDeclarationOnClass(
-                        pCurDeclMethod,
-                        pCurImplMethod);
-                }
-                else
+                // Don't place static virtual method overrides in the vtable
+                if (!IsMdStatic(pCurDeclMethod->GetDeclAttrs()))
```

Comment:
This check should be immediately after the hDeclMethod.AsRTMethod() call. Also, it should be duplicated for the case above where the pCurDeclMethod is a MDMethod.

Reply:
Fixed in 4th comment - I believe we can make the entire section after retrieval of <code>bmtMethodHandle hDeclMethod</code> conditional on non-static-ness.